core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/audio/sound/BasicSoundPlayer.java

```diff
@@ -38,14 +40,18 @@
  */
 public class BasicSoundPlayer implements Runnable, SoundPlayer {
 
-	private QueuedInputStream stream;
-	private AudioInputStream audioStream;
-	private byte[] data;
-	private int pos;
-	private int remainingLoopCount;
+	private final QueuedInputStream stream;
+	private final AudioInputStream audioStream;
+
+	private volatile boolean shouldStop = false;
+	private volatile byte[] data;
+	private volatile int pos;
+	private volatile int remainingLoopCount;
 
 	private Thread playingThread;
 
+	private Consumer<SoundPlayer.StopReason> listener;
```

Comment:
I think strictly speaking this should be volatile too? Or alternatively final, as a constructor parameter?

Reply:
Actually the variable is effectively final in this context, because it can't be altered after start() has been called. And a new Thread creates the local copy of all variables when start() is called. So it does not need to be volatile.